spec/inputs/jdbc_spec.rb

```diff
@@ -113,4 +114,59 @@
 
     @database.drop_table(:test_table)
   end
+
+  context "persistence" do
+
+    it "should respect last run metadata" do
+      settings = {
+        "statement" => "SELECT * FROM SYSIBM.SYSDUMMY1",
+        "last_run_metadata_path" => Stud::Temporary.pathname
+      }
+      plugin = LogStash::Inputs::Jdbc.new(mixin_settings.merge(settings))
+      plugin.register
+      q = Queue.new
+      plugin.run(q)
+      plugin.teardown
+
+      insist { File.exists?(settings["last_run_metadata_path"]) }
+      last_run = YAML.load(File.read(settings["last_run_metadata_path"]))
+
+      plugin = LogStash::Inputs::Jdbc.new(mixin_settings.merge(settings))
+      plugin.register
+      insist { plugin.instance_variable_get("@sql_last_start") } == last_run
+      plugin.teardown
+    end
+
+    it "should ignore last run metadata if :clean_run set to true" do
+      settings = {
+        "statement" => "SELECT * FROM SYSIBM.SYSDUMMY1",
+        "last_run_metadata_path" => Stud::Temporary.pathname,
+        "clean_run" => true
+      }
+
+      File.open(settings["last_run_metadata_path"], "w") { |f| f.write(YAML.dump(Time.at(1).utc)) }
```

Comment:
it would be awesome if we could standardize a way between plugins. should we add such a module to logstash-core? (unless it already exists and I missed it)